merlin/systems/dag/ops/pytorch.py

```diff
@@ -0,0 +1,191 @@
+        selector: ColumnSelector,
+    ) -> Schema:
+        """
+        Use the input schema supplied during object creation.
+        """
+        return self.input_schema
+
+    def compute_output_schema(
+        self, input_schema: Schema, col_selector: ColumnSelector, prev_output_schema: Schema = None
+    ) -> Schema:
+        """
+        Use the output schema supplied during object creation.
+        """
+        return self.output_schema
+
+    def export(self, path, input_schema, output_schema, node_id=None, version=1):
+        """Create a directory inside supplied path based on our export name"""
+        node_name = f"{node_id}_{self.export_name}" if node_id is not None else self.export_name
+
+        node_export_path = pathlib.Path(path) / node_name
+        node_export_path.mkdir(exist_ok=True)
+
+        export_model_path = pathlib.Path(node_export_path) / str(version)
+        export_model_path.mkdir(exist_ok=True)
+
+        if self.path:
+            copyfile(
+                str(self.path),
+                export_model_path / "model.pt",
+            )
+        else:
+            self.model.save(export_model_path / "model.pt")
+
+        return self._export_model_config(node_name, node_export_path)
+
+    def _export_model_config(self, name, output_path):
+        """Exports a PyTorch model for serving with Triton
+
+        Parameters
+        ----------
+        name:
+            The name of the triton model to export
+        output_path:
+            The path to write the exported model to
+        """
+        config = self._export_torchscript_config(name, output_path)
+        # TODO: Add support for Python back-end configs here
+
+        return config
+
+    def _export_torchscript_config(self, name, output_path):
+        """Exports a PyTorch model for serving with Triton
+
+        Parameters
+        ----------
+        name:
+            The name of the triton model to export
+        output_path:
+            The path to write the exported model to
+        """
+        config = model_config.ModelConfig(name=name)
+
+        config.backend = "pytorch"
+        config.platform = "pytorch_libtorch"
+        config.parameters["INFERENCE_MODE"].string_value = "true"
+
+        for col_name, col_schema in self.input_schema.column_schemas.items():
+            dims = [-1, 1]
+
+            if col_schema.is_list and not col_schema.is_ragged:
+                value_count = col_schema.properties.get("value_count", None)
+                if value_count and value_count["min"] == value_count["max"]:
+                    dims = [-1, value_count["max"]]
+
+            config.input.append(
+                model_config.ModelInput(
+                    name=col_name, data_type=_convert_dtype(col_schema.dtype), dims=dims
+                )
+            )
+
+        for col_name, col_schema in self.output_schema.column_schemas.items():
+            # this assumes the list columns are 1D tensors both for cats and conts
```

Comment:
This comment is non-blocking for this PR. I'm wondering if there are any ways to inspect the output shape you considered but didn't implement. (e.g. running a forward pass through the model with synthetic data?) to allow us to remove this restriction. As we have discussed before, maybe only a priority if this assumption is something that is easy to break with Merlin Models.

Reply:
I suppose related to this TODO comment
https://github.com/NVIDIA-Merlin/systems/blob/905143cb0a695abc7f94f9c69bb81fd31f46b802/merlin/systems/dag/ops/pytorch.py#L66

Reply:
Yeah, there are definitely issues outstanding here; I would not recommend trying to use this for anything real yet. Mostly want to get the preliminary support merged so that it's easier to test out with the Torchscript versions of Transformers4Rec models. I expect to find a lot of further issues when we do that.